static/app/stores/projectsStore.tsx

```diff
@@ -78,13 +76,17 @@ const storeConfig: ProjectsStoreDefinition = {
     }
 
     const newProject = {...prevProject, slug: newSlug};
+    this.projects = this.projects
+      .map(project => (project.slug === prevSlug ? newProject : project))
+      .sort((a, b) => a.slug.localeCompare(b.slug));
 
-    this.itemsById = {...this.itemsById, [newProject.id]: newProject};
     this.trigger(new Set([prevProject.id]));
   },
 
   onCreateSuccess(project: Project, orgSlug: string) {
-    this.itemsById = {...this.itemsById, [project.id]: project};
+    this.projects = [...this.projects, project].sort((a, b) =>
```

Comment:
nit: this.projects.concat(project) avoid the extra iterator call

Reply:
i always forget that it returns a new array